host_core/lib/host_core/web_assembly/imports.ex

```diff
@@ -299,15 +300,83 @@ defmodule HostCore.WebAssembly.Imports do
       )
       |> perform_rpc_invoke(target_subject)
 
-    case unpack_invocation_response(invocation_res) do
-      {1, :host_response, msg} ->
-        Agent.update(agent, fn state -> %State{state | host_response: msg} end)
-        1
+    unpacked = unpack_invocation_response(invocation_res)
 
-      {0, :host_error, error} ->
-        Agent.update(agent, fn state -> %State{state | host_error: error} end)
-        0
-    end
+    res =
+      case unpacked do
+        {1, :host_response, msg} ->
+          Agent.update(agent, fn state -> %State{state | host_response: msg} end)
+          1
+
+        {0, :host_error, error} ->
+          Agent.update(agent, fn state -> %State{state | host_error: error} end)
+          0
+      end
+
+    Task.start(fn ->
+      publish_invocation_result(
+        actor,
+        namespace,
+        binding,
+        operation,
+        byte_size(payload),
+        target_type,
+        target_key,
+        res
+      )
+    end)
+
+    res
+  end
+
+  defp publish_invocation_result(
+         actor,
+         namespace,
+         binding,
+         operation,
+         payload_bytes,
+         target_type,
+         target_key,
+         res
+       ) do
+    evt_type =
+      if res == 1 do
+        "invocation_succeeded"
+      else
+        "invocation_failed"
+      end
+
+    prefix = HostCore.Host.lattice_prefix()
+
+    msg =
+      %{
```

Comment:
This large nested map structure shows up in 2 places, I think it would be cleaner if we had a helper function that could take all of the parameters and form the structure just to avoid having to change it in both places.

Reply:
yeah I'd refactored this thing a bunch of times, so there's some leftover goo.

Reply:
so I agree that it would ordinarily be a good idea to collapse this into one.. but when you look at the sources of data for each of the two uses of this structure, they're very different. By the time I massage both inputs into a central way of producing this structure, it ends up being more code than the 2 separate uses.

Reply:
Yeah that's fair, I'm fine with just leaving as-is. We'll just have to keep that in mind for any future updates to the invocation event structure